src/Microsoft.Dism/NativeConstants.cs

```diff
@@ -21,7 +21,7 @@ public static partial class DismApi
         /// <summary>
         /// Indicates to the DismUnmountImage Function that all changes should be saved. This flag is equivalent to using DISM_COMMIT_IMAGE, DISM_COMMIT_GENERATE_INTEGRITY, and DISM_COMMIT_APPEND.
         /// </summary>
-        public const uint DISM_COMMIT_MASK = 0xFFFF0000;
+        public const uint DISM_COMMIT_MASK = DISM_COMMIT_IMAGE | DISM_COMMIT_GENERATE_INTEGRITY | DISM_COMMIT_APPEND;
```

Comment:
Sorry for the long delay, things have been hectic at work.  

I think the correct fix is to pass `DISM_COMMIT_IMAGE` in `CommitImage`

https://github.com/josemesona/ManagedDism/blob/master/src/Microsoft.Dism/DismApi.cs#L234

```diff
-UInt32 flags = discardChanges ? DismApi.DISM_DISCARD_IMAGE : DismApi.DISM_COMMIT_MASK;
+UInt32 flags = discardChanges ? DismApi.DISM_DISCARD_IMAGE : DismApi.DISM_COMMIT_IMAGE;
```

In `Dism.h`, `DISM_COMMIT_MASK` is set to `0xFFFF0000` so I think the value here is correct.  Its just the flag passed to `DismCommitImage` that needs to be changed

```C++
// Commit flags may also be used with unmount.  AND this with unmount flags and you will
// get the commit-specific flags.
#define DISM_COMMIT_MASK                0xffff0000
```

[dismapi.h.txt](https://github.com/josemesona/ManagedDism/files/1456252/dismapi.h.txt)